specification/compute/resource-manager/Microsoft.Compute/stable/2021-07-01/compute.json

```diff
@@ -9610,10 +9613,30 @@
             "name": "VirtualMachineSizeTypes",
             "modelAsString": true
           }
+        },
+        "vmSizeProperties": {
+          "$ref": "#/definitions/VMSizeProperties",
+          "description": "Specifies the VM Size Properties for customizing VM sizes."
         }
       },
       "description": "Specifies the hardware settings for the virtual machine."
     },
+    "VMSizeProperties": {
+      "type": "object",
+      "properties": {
+        "vCPUsAvailable": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Indicates the number of vCPUs available for the VM. <br><br> When this property is not specified in the request body, default behavior is to set it to the default number of vCPUs available for that VM size."
```

Comment:
Specifies the number of vCPUs available for the VM. <br><br> When this property is not specified in the request body, the default behavior is to set it to the value of vCPUs available for that VM size exposed in api response of  [List all available virtual machine sizes in a region]( https://docs.microsoft.com/rest/api/compute/resourceskus/list)

Reply:
ack